### PR TITLE
Get PK on other chains

### PR DESCRIPTION
**Pull request by apaillier-ledger**
## Description

Part of the EVM framework changes.

* Aligns the NBGL flow with other apps
* Get PK Ragger tests

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [x] Tests
- [x] Documentation
- [ ] Other (for changes that might not fit in any category)

### CODE REVIEW

File: src_nbgl/ui_get_public_key.c [hunk 1]
```diff
@@ -2,27 +2,54 @@
...
 
-static void buildScreen(void) {
-    nbgl_useCaseAddressConfirmation(strings.common.fullAddress, reviewChoice);
+static void display_addr(void) {
+    nbgl_useCaseAddressConfirmation(strings.common.fullAddress, review_choice);
+}
+
+void ui_display_public_key(const uint64_t *chain_id) {
+    strlcpy(g_stax_shared_buffer, "Verify ", sizeof(g_stax_shared_buffer));
+    if (chain_id != NULL) {
```
Comment (fbeutin-ledger) [inline src_nbgl/ui_get_public_key.c:38]: Add a comment

File: src_nbgl/ui_get_public_key.c [hunk 2]
```diff
@@ -2,27 +2,54 @@
...
+        if (chain_is_ethereum_compatible(chain_id)) {
+            strlcat(g_stax_shared_buffer,
+                    get_network_name_from_chain_id(chain_id),
+                    sizeof(g_stax_shared_buffer));
+            strlcat(g_stax_shared_buffer, "\n", sizeof(g_stax_shared_buffer));
```
Comment (fbeutin-ledger) [inline src_nbgl/ui_get_public_key.c:43]: What is the longest possible length of `get_network_name_from_chain_id(chain_id),` ?
Is it possible that we end up with
```
"Verify SuperLongNet
work
address"
```
Due to auto wrapping ?

Comment (fbeutin-ledger) [reply src_nbgl/ui_get_public_key.c:43]: If yes, calculate `len(get_network_name_from_chain_id(chain_id))` first and place the `\n` accordingly

Comment (fbeutin-ledger) [reply src_nbgl/ui_get_public_key.c:43]: This should be done by the SDK .....

File: src_bagl/ui_network.c [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#include "ui_network.h"
+#include "shared_context.h"
+
+//////////////////////////////////////////////////////////////////////
+// clang-format off
+UX_STEP_NOCB(
```
[the remark this answers is not in the document]
Comment (apaillier-ledger) [reply src_bagl/ui_network.c]: Initially, this UX_FLOW was also used for the get_public_key for Nanos so it was extracted from the signing code.
Now that this is not the case anymore, theses changes have been reverted. :+1:

File: src_features/getPublicKey/cmd_getPublicKey.c [hunk 4]
```diff
@@ -51,12 +52,31 @@ void handleGetPublicKey(uint8_t p1,
...
+
+        dataLength -= sizeof(chain_id);
+        dataBuffer += sizeof(chain_id);
+    }
+
+    if (dataLength > 0) {
+        THROW(APDU_RESPONSE_INVALID_DATA);
+    }
+
 #ifndef NO_CONSENT
```
[the remark this answers is not in the document]
Comment (apaillier-ledger) [reply src_features/getPublicKey/cmd_getPublicKey.c:76]: To check for unwanted leftover data in the APDU payload.

File: src_nbgl/ui_get_public_key.c (hunk 2 above)
Comment (fbeutin-ledger) [reply src_nbgl/ui_get_public_key.c:46]: Feels bad... No way to calculate the length and place the '\n' accordingly ? I've done something similar in Exchange when we display the appname of the lib

**Review by fbeutin-ledger — APPROVED** [review]

**Review by fbeutin-ledger — APPROVED** [review]

**Pull request merged**